### ISSUE TITLE
GCC Native Ext. Build Not Working on Windows

### ISSUE THREAD

**Issue by piboistudios** [Aurel300/ammer#30]
I'm really not sure what domain this problem falls within or if any (maybe something I can fix on my own), but I have a native library,

take these two Makefiles:
Makefile.win:
```
all: odbc.dll
    @:

odbc.dll: odbc.obj
    gcc -shared -mwindows -o odbc.dll odbc.obj -lodbc32

odbc.obj: odbc.c
    gcc -c -mwindows odbc.c -o odbc.obj -lodbc32

.PHONY: all
```
and Makefile.msvc:
```
all: odbc.dll
    @:

odbc.dll: odbc.obj
    cl /LD odbc.obj /link /OUT:odbc.dll /LIBPATH:"C:\Program Files (x86)\Windows Kits\10\Lib\10.0.17763.0\um\x86" odbc32.lib

odbc.obj: odbc.c
    cl /c odbc.c 

.PHONY: all
```

Both of them work, they both produce "odbc.dll"; Makefile.msvc also produces `exp` and `lib` files.

For the GCC approach, I used `dumpfile /EXPORTS odbc.dll > odbc.exports` and created a DEF file to then create a LIB/EXP file from using the Microsoft `lib` tool.

This works for GCC, so end result after building I end up with the same files:
- odbc.dll
- odbc.lib
- odbc.exp

The problem is, odbc.dll cannot be loaded by ammer_odbc.hdll (presumably that's where this failure happens). I get this error: `src\module.c(321) : FATAL ERROR : Failed to load library ammer_odbc.hdll`

Does anyone have experience using GCC built DLLs with Ammer in Windows? I know other platforms require GCC, so I imagine it has to be possible on windows as well.

**Comment by piboistudios** [comment]
I'm a C/C++ rookie, in this case I am going to assume that because I installed HL via binaries, it was probably built via the MSVC toolchain. 

If I built it via GCC I imagine this should work. Anywho, closing for now.

**Comment by Aurel300** [comment]
I don't think that a different compiler framework should cause a DLL to not work properly. However, a version mismatch might be the cause? Anyway, I think these types of issues might be resolved when I get CI setup (#24).

**Comment by piboistudios** [comment]
Are you also working on a build tool for library authors?

I'm in the process of creating a macro that gets called from extraParams.hxml in a `--macro` call to build and copy the native files to the user include/output directories whenever they build.

It would take in the native path and ammer lib name and build cmd as input params:
- copy the native path from the library on user computer to the user project directory (into a subdirectory for ammer includes)
- run the build command
- use the ammer lib name to define an `ammer.lib.<ammerlib>.include` to the directory where the native source files were copied (and where it's built output is)
- based on platform, copy dll/so/dylib to the user output directories

This would then allow ammer lib authors to use a simple config to have their projects automatically bootstrapped into users projects. 

Conceptually, it should work independent of the package manager used (whether it's haxelib, lix or hmm).

Sound useful?
